src/components/PopupWithForm.js

```diff
@@ -27,10 +28,38 @@ class PopupWithForm extends Popup {
     super.setEventListeners();
     this._form.addEventListener('submit', (e) => {
       e.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this._form.reset(); // Clear inputs after form submission
+      this.renderLoading(true);
+
+      const submitResult = this._handleFormSubmit(this._getInputValues());
+      
+      if (submitResult && typeof submitResult.then === 'function') {
+        submitResult
+          .then(() => {
```

Comment:
Please, delete `then`, `catch`, `finally` from `PopupWithForm` because you do everything in `index.js`